go/libraries/doltcore/sqle/dolt_index.go

```diff
@@ -273,3 +274,59 @@ func (di *doltIndex) keysToTuple(keys []interface{}) (types.Tuple, error) {
 	}
 	return types.NewTuple(nbf, vals...)
 }
+
+func (di *doltIndex) Equals(oIdx DoltIndex) bool {
+	if !expressionsAreEquals(di.Expressions(), oIdx.Expressions()) {
+		return false
+	}
+
+	if !(di.Database() == oIdx.Database()) {
+		return false
+	}
+
+	if !(di.ID() == oIdx.ID()) {
```

Comment:
Pretty sure we generate unique IDs for indexes, so you can probably just check this

Reply:
I see what you're saying